apple/REAModule.mm

```diff
@@ -158,47 +159,35 @@ - (void)handleJavaScriptDidLoadNotification:(NSNotification *)notification
 
 /*
  * Taken from RCTNativeAnimatedTurboModule:
- * In bridgeless mode, `setBridge` is never called during initialization. Instead this selector is invoked via
- * BridgelessTurboModuleSetup.
+ * This selector is invoked via BridgelessTurboModuleSetup.
  */
 - (void)setSurfacePresenter:(id<RCTSurfacePresenterStub>)surfacePresenter
 {
   _surfacePresenter = surfacePresenter;
+  _isBridgeless = true;
 }
 
 - (void)setBridge:(RCTBridge *)bridge
 {
-  // This method isn't called on Bridgeless mode.
   [super setBridge:bridge];
-
-  [bridge.uiManager.observerCoordinator addObserver:self];
```

Comment:
We don't use it on the new architecture

Reply:
1. I was wondering if we use it in RNGH to pass events to Reanimated but turns out we don't (in RNGH there's a function `sendEventForReanimated` that on new arch calls `reanimatedEventDispatcher.sendEvent()`  which in turn calls `reanimatedModule?.nodesManager?.onEventDispatch(event)` so I guess that's okay).
2. If it's not needed on the new architecture, let's wrap `<RCTEventDispatcherObserver>` in header file and `eventDispatcherWillDispatchEvent:` method inside `#ifdef RCT_NEW_ARCH_ENABLED`, what do you think?

Reply:
As we discussed offline, we refactor the whole initialization process in the following PR.